snekbox/nsjail.py

```diff
@@ -227,13 +239,14 @@ def python3(
             # Parse attachments
             try:
                 # Sort attachments by name lexically
-                attachments = sorted(
-                    fs.attachments(self.max_attachments, self.max_attachment_size),
-                    key=lambda a: a.name,
+                attachments = timed(
+                    parse_files,
+                    (fs, self.files_limit, self.files_pattern),
+                    timeout=self.files_timeout,
                 )
                 log.info(f"Found {len(attachments)} attachments.")
-            except AttachmentError as err:
-                log.info(f"Failed to parse attachments: {err}")
+            except TimeoutError as err:
+                log.info(f"Exceeded time limit in parsing attachments: {err}")
```

Comment:
I have mixed feelings towards having a timeout. I'm leaning towards keeping it simple and not having a timeout. It should be possible to prevent long parsing times by just limiting the amount of files, for which we already have a setting.

Furthermore, it seems unintuitive that the process may have an OK exit code with some stdout available, but it's getting overwritten by an error, and the HTTP response is still 200. This is also a concern I had with the previous code that was erroring out when a file size limit was exceeded.

What are everyone's thoughts on having a timeout for file parsing? Is it beneficial given the other measures we have in place?

Reply:
My main concern is, if we allow nested directories in outputs, someone could create a very deeply nested directory as a denial of service attack. 

In general this is a potentially lengthy part of server side code execution that can be entirely controlled by the user.

Reply:
I agree that the response feels unintuitive though. Maybe it can just be a 500 error? This is more of a catastrophic fail-safe than something that is going to come up often.

Reply:
I have changed the returncode of the TimeoutError to `None` to be in-line with the previous snekbox stdout errors.